server/api.go

```diff
@@ -45,12 +46,56 @@ type PRDetails struct {
 	Reviews            []*github.PullRequestReview `json:"reviews"`
 }
 
+type HTTPHandlerFuncWithUser func(w http.ResponseWriter, r *http.Request, userID string)
+
 func writeAPIError(w http.ResponseWriter, err *APIErrorResponse) {
 	b, _ := json.Marshal(err)
 	w.WriteHeader(err.StatusCode)
 	w.Write(b)
 }
 
+func (p *Plugin) initialiseAPI() {
+	p.router = mux.NewRouter()
+
+	webHookRouter := p.router.PathPrefix("/webhook").Subrouter()
+	oauthRouter := p.router.PathPrefix("/oauth").Subrouter()
+	apiRouter := p.router.PathPrefix("/api/v1").Subrouter()
+
+	webHookRouter.HandleFunc("", p.handleWebhook).Methods("POST")
```

Comment:
With this change, is a subrouter really needed?

Reply:
No, I think it's redundant now. I'll change that